scylla-cdc-base/src/main/java/com/scylladb/cdc/model/worker/Worker.java

```diff
@@ -100,13 +129,25 @@ private TaskActionsQueue queueFirstActionForEachTask(Map<TaskId, SortedSet<Strea
      * At each iteration, runs a single action from |actions| queue if any
      * available.
      */
-    private void performActionsUntilStopRequested(TaskActionsQueue actions) {
-        while (!workerConfiguration.transport.shouldStop()) {
-            try {
-                actions.runNextAction();
-            } catch (InterruptedException e) {
-                // Ignore InterruptedException
-            }
+    @SuppressWarnings("deprecation")
+    private void performActionsUntilStopRequested(Collection<TaskAction> actions) {
+        if (shouldStop()) {
+            return;
+        }
+
+        ScheduledExecutorService executorService = getExecutorService();
+        try {
+            executorService.invokeAll(actions.stream().map(a -> makeCallable(a)).collect(Collectors.toSet()));
+            do {
+                // pretty short poll, to allow for reasonably fast switchover
+                // iff using "polled" shutdown (WorkerTransport::shouldStop).
+                executorService.awaitTermination(50, MILLISECONDS);
+                if (workerConfiguration.transport.shouldStop()) {
```

Comment:
After the changes below this always returns false. I'm a bit confused.

Reply:
I mean I understand why the code works but I don't like the fact that the local transport diverges so much from the kafka connect one. It was simpler when they are similar.

Reply:
There is nothing inherent in the Kafka transport that it could not call explicit "stop" instead of setting a status thing. I did not rewrite it, because this is a big enough PR as it is, and something that checks backwards compat during switchover is good. 

But the point is still that calling explicit "stop" is gonna be faster that eventually returning true in "shouldStop", since the latter is dependent on polling here or in task queues.

Reply:
That's not true as far as I understand. Master does not have handles to tasks created by the framework and can't just stop them. Maybe my understanding is wrong. @avelanarius please comment.